### PR TITLE
Fixed issue #629, Added Vibration preferences and controls where used

### PR DESCRIPTION
**Pull request by smarek**
This proposal is to allow users to control the vibration when status changes and to set threshold on which phone will vibrate

### CODE REVIEW

**Comment by SecUpwN** [comment]
Thank you so much for your effort to improve our code, @smarek! As you can see, we've had a very long downtime due to my life changing pretty rapidly, I'm still fighting my way back online and hope to bring back our project to life and blossom soon. @He3556 and @DJaeger would you please review this PR?

**Comment by agilob** [comment]
Review done. Fix awaiting.

File: app/src/main/java/com/SecUpwN/AIMSICD/adapters/AIMSICDDbAdapter.java [hunk 1]
```diff
@@ -138,10 +142,12 @@
     private String[] mTables;
     private SQLiteDatabase mDb;
     private Context mContext;
+    private SharedPreferences mPreferences;
 
     public AIMSICDDbAdapter(Context context) {
         super(context, DB_NAME, null, 1);
         mContext = context;
+        mPreferences = PreferenceManager.getDefaultSharedPreferences(mContext);
```
Comment (agilob) [inline app/src/main/java/com/SecUpwN/AIMSICD/adapters/AIMSICDDbAdapter.java]: Is there a reason to make this field class global? It's used only in one method(?)

Comment (smarek) [inline app/src/main/java/com/SecUpwN/AIMSICD/adapters/AIMSICDDbAdapter.java]: Yes, sorry, missed that this is constructor and not public method. Will fix in moments

Comment (smarek) [inline app/src/main/java/com/SecUpwN/AIMSICD/adapters/AIMSICDDbAdapter.java]: Yes actually, this method is called heavily, so creating new instance in this method will be more resource ineffective than keeping the SharedPreferences reference in memory.

Comment (agilob) [inline app/src/main/java/com/SecUpwN/AIMSICD/adapters/AIMSICDDbAdapter.java]: It doesn't matter how often the constructor is called.  

`private SharedPreferences mPreferences;` it's not static final so the object will be reconstructed every time a compiler/method is called. Field can be limited to method-body or classifier changer to static final to limit resource usage.

Comment (smarek) [inline app/src/main/java/com/SecUpwN/AIMSICD/adapters/AIMSICDDbAdapter.java]: No, in constructor is the field filled, however it's used in method `toEventLog` (line 2172), which is called from various locations. final modifier would be ok imho.

File: app/src/main/java/com/SecUpwN/AIMSICD/adapters/AIMSICDDbAdapter.java [hunk 2]
```diff
@@ -2161,9 +2167,18 @@ public void toEventLog(int DF_id, String DF_desc){
 
                 mDb.insert("EventLog", null, eventLog);
                 Log.i(TAG, "ToEventLog(): Added new event: id=" + DF_id + " time=" + time + " cid=" + cid);
+
                 // Short 100 ms Vibration
-                Vibrator v = (Vibrator) mContext.getSystemService(Context.VIBRATOR_SERVICE);
-                v.vibrate(100);
+                // TODO not elegant solution, vibrator invocation should be moved somewhere else imho
+                Log.d(TAG, String.format("vibrate enabled:%b",mContext.getString(R.string.pref_notification_vibrate_enable)));
```
Comment (agilob) [inline app/src/main/java/com/SecUpwN/AIMSICD/adapters/AIMSICDDbAdapter.java]: I know it's pedantic and very minor... but String.format is a really, really, slow, slow method, and you should use plus operator here, which is converter to stringbuilder, that's faster way, or do:
 `"vibrate enabled:%b"`.replace("%d"...

I think `+` is more preferable. I know, it's log.debug and minor but string.format should be avoided on android. sorry for spotting this late.

File: app/src/main/java/com/SecUpwN/AIMSICD/adapters/AIMSICDDbAdapter.java (hunk 1 above)
Comment (agilob) [inline app/src/main/java/com/SecUpwN/AIMSICD/adapters/AIMSICDDbAdapter.java]: the constructor is called a lot of times in the code base, see yourself ;)

Comment (smarek) [inline app/src/main/java/com/SecUpwN/AIMSICD/adapters/AIMSICDDbAdapter.java]: IMHO using static references to non-static context bound variables is memory leak by definition. Correct me if I'm wrong.

File: app/src/main/java/com/SecUpwN/AIMSICD/adapters/AIMSICDDbAdapter.java (hunk 2 above)
Comment (smarek) [inline app/src/main/java/com/SecUpwN/AIMSICD/adapters/AIMSICDDbAdapter.java]: I'm sorry, I shouldn't have left the Log call in there, it was just for debugging purpose, will remove it.

**Comment by SecUpwN** [comment]
Thanks for checking this PR throughly, @agilob. I am aiming at crafting a fresh release this sunday, I hope that this bugfix will make it into it - if @smarek is able to change his lovely code accordingly. ;-)

File: app/src/main/java/com/SecUpwN/AIMSICD/adapters/AIMSICDDbAdapter.java (hunk 1 above)
Comment (agilob) [inline app/src/main/java/com/SecUpwN/AIMSICD/adapters/AIMSICDDbAdapter.java]: You're right, didn't notice that, sorry o/

**Comment by smarek** [comment]
@SecUpwN I've updated PR with strings fixed for translation purposes. If I've missed anything, I'll do it right away. Thanks for the review @agilob

**Pull request merged**